### PR TITLE
[TF FE] Support GRUBlockCell

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Currently, we support only hidden state output from GRUBlockCell due to OpenVINO GRUCell capability

**Ticket:** 92653

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

### CODE REVIEW

File: src/frontends/tensorflow/src/helper_ops/gru_block_cell.hpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            FRONT_END_OP_CONVERSION_CHECK(
+                b_c_rank.get_length() == 1,
+                "Internal error in OpenVINO TensorFlow Frontend: bias for GRUBlockCell must be of rank equal to 1.");
+            m_hidden_size = b_c_shape[0].is_static() ? b_c_shape[0].get_length() : m_hidden_size;
+        }
+
+        // set the defined shape only for the fourth output since
+        // OpenVINO GRUCell supports hidden state output
+        auto x_type = get_input_element_type(0);
+        set_output_type(0, x_type, ov::PartialShape::dynamic());
```
Comment (mvafin) [inline src/frontends/tensorflow/src/helper_ops/gru_block_cell.hpp:98]: Can we at least specify rank for these three outputs if the shape is dynamic?

Comment (mvafin) [reply src/frontends/tensorflow/src/helper_ops/gru_block_cell.hpp:98]: I see it is not used, okay in that case

File: src/frontends/tensorflow/src/helper_ops/gru_block_cell.hpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            m_hidden_size = w_c_shape[1].is_static() ? w_c_shape[1].get_length() : m_hidden_size;
+        }
+        // 3. use b_ru shape
+        auto b_ru_shape = get_input_partial_shape(4);
+        auto b_ru_rank = b_ru_shape.rank();
+        if (b_ru_rank.is_static()) {
+            FRONT_END_OP_CONVERSION_CHECK(
+                b_ru_rank.get_length() == 1,
+                "Internal error in OpenVINO TensorFlow Frontend: bias for GRUBlockCell must be of rank equal to 1.");
+            m_hidden_size = b_ru_shape[0].is_static() ? b_ru_shape[0].get_length() / 2 : m_hidden_size;
```
Comment (itikhono) [inline src/frontends/tensorflow/src/helper_ops/gru_block_cell.hpp:83]: what if b_ru_shape[0].get_length() / 2 is not equal the current value of m_hidden_size?
Maybe we need to add an additional check in every if

File: src/frontends/tensorflow/src/helper_transforms/gru_block_cell_replacer.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+#include "utils.hpp"
+
+using namespace std;
+using namespace ov::pass;
+using namespace ov::opset9;
+using namespace ov::frontend::tensorflow;
+
+pass::GRUBlockCellReplacer::GRUBlockCellReplacer() {
```
Comment (itikhono) [inline src/frontends/tensorflow/src/helper_transforms/gru_block_cell_replacer.cpp:24]: please add test

File: src/frontends/tensorflow/src/helper_transforms/gru_block_cell_replacer.cpp [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        auto b_c = gru_block_cell_node->input_value(5);
+
+        // retrive input_size and hidden_size
+        auto x_shape = rg.make<ShapeOf>(x, element::i64);
+        auto ss_start = rg.make<Constant>(element::i64, Shape{1}, 1);
+        auto ss_end = rg.make<Constant>(element::i64, Shape{1}, 2);
+        auto ss_step = rg.make<Constant>(element::i64, Shape{1}, 1);
+        auto input_size = rg.make<Slice>(x_shape, ss_start, ss_end, ss_step);
+        auto h_prev_shape = rg.make<ShapeOf>(h_prev, element::i64);
+        auto hidden_size = rg.make<Slice>(h_prev_shape, ss_start, ss_end, ss_step);
```
Comment (itikhono) [inline src/frontends/tensorflow/src/helper_transforms/gru_block_cell_replacer.cpp:64]: could you check case where hidden_size and input_size are dynamic in input shape (h_prev_shape)?
please add such a test

File: src/frontends/tensorflow/src/helper_transforms/gru_block_cell_replacer.cpp (hunk 3 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/helper_transforms/gru_block_cell_replacer.cpp:24]: Added

File: src/frontends/tensorflow/src/helper_ops/gru_block_cell.hpp (hunk 2 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/helper_ops/gru_block_cell.hpp:83]: Discussed offline

**Review by itikhono — APPROVED** [review]

**Review by mvafin — APPROVED** [review]

**Pull request merged**